Admin theme re-design.

xSido opened the pull request:
Admin theme re-design.

Login page:
-removed top help link as it already could be found in the bottom one.
-moved go to main site link to the right.
-edited css to simplify the mess with positioning.
-made a new login button , with a little animation when clicked.
-added the themed blue line.

Admin panel:
-left menu is now hidden for a cleaner look.
-added dashboard link to the menu.
-top settings link is now settings only, and does not include theme settings anymore.
-removed the top navigation buttons for a cleaner look.
-added order/invoice/support buttons to the top menu with notifications.
-moved profile avatar to the right of the top menu.
-clicking profile name will also redirect to profile page now.
-edited display name for client in certain tables to show entire first name up to 15 characters, and then last name initial only. When a client register, it is asked for his first name only, therefore most of the clients will not update in profile their last name too and this will result into having your statistic tables filled up only with first name initials.
-edited display stats look.
-added the themed blue line.

CSS:
-minified to beautify.

NOTE:
-i haven't touched/checked if is responsive to different resolutions.
-expecting reports if any problem was introduced.

BenNottelling commented:
Hey, @xSido 
Would you mind attaching some screenshots of the new look?

Thank you!

xSido commented:
> Hey, @xSido Would you mind attaching some screenshots of the new look?
> 
> Thank you!

Ofc not. Here they are.
![4](https://user-images.githubusercontent.com/45123194/144349997-1c513278-1430-4ffe-86ac-77f217a5a30b.png)
![awdwadawdawd](https://user-images.githubusercontent.com/45123194/144350001-905305c0-ec6b-4dee-806e-4cf48eaaea3f.png)
![2](https://user-images.githubusercontent.com/45123194/144350005-87485b3e-7270-41c1-abd4-fb0de08af80f.png)
![1](https://user-images.githubusercontent.com/45123194/144350007-073b7c42-fdab-47a8-98c8-7c19ca8932ba.png)

BenNottelling reviewed (commented):
Overall, looks very good and I'd love to have it merged in! 
A couple minor things I'd appreciate you look at:
I would like it if the entire thing here acted as the link, rather than just the numbers
![image](https://user-images.githubusercontent.com/17304943/144372235-a42981c8-b9d0-4922-849c-972ac1e20478.png)

> i haven't touched/checked if is responsive to different resolutions.

I tested it and the theme definitely isn't responsive. Mostly fine on larger screens (ipad or bigger probably), but not phones. It becomes unusable on a phone because the menu can no longer be expanded.
I doubt the original was to begin with either, but it's something we should work on.
With that being said, I'm not going to make it a requirement to merge this.

Also, it's pretty minor but I think we should keep "min.css" minimized for production.
Would be perfectly fine with you creating a new one that isn't minimized, but keeping it minimized is best practice for production.

Thank you again for the theme update!

BenNottelling commented:
I just pushed a slight change to center these icons
![image](https://user-images.githubusercontent.com/17304943/144378300-7394ca53-e657-4496-96f5-e21d91e23e4a.png)

Also, It's not work that you did, but I'm not sure if we should have the orders icon link to orders that have failed setup

xSido commented:
> : I would like it if the entire thing here acted as the link, rather than just the numbers ![image](https://user-images.githubusercontent.com/17304943/144372235-a42981c8-b9d0-4922-849c-972ac1e20478.png)
> 
yeah, thought the same, with next update :D

> > i haven't touched/checked if is responsive to different resolutions.
> I tested it and the theme definitely isn't responsive. Mostly fine on larger screens (ipad or bigger probably), but not phones. It becomes unusable on a phone because the menu can no longer be expanded.
> 
that will be the next step 

> Also, it's pretty minor but I think we should keep "min.css" minimized for production. Would be perfectly fine with you creating a new one that isn't minimized, but keeping it minimized is best practice for production.
> 
i have no problem in doing that.

evrifaessa reviewed (changes requested):
Looks nice except the togglable sidebar. I think it can be improved but I like the idea.

xSido commented:
> Looks nice except the togglable sidebar. I think it can be improved but I like the idea.

in future will be improved (design speaking) but atm is the base and need to be functional :D

xSido commented:
Admin theme re-design - fix responsive.

Login page:
-fixed go to main site link (it was disappearing on mobile).
-added space under login panel (bottom was not visible under scroll).
-footer is now fixed (it was scrolling with the page).

Admin panel:
-top menu is now hided into a right panel in mobile view.
-fixed left menu submenu (were not responsive).
-fixed stats (were not responsive).
-reduced size of side menues.





Admin theme re-design - fix stats.

Stats:
-fixed size.
-fixed responsive size.
-entire list is now a link (istead just the number).

NOTE:
-expecting reports if any problem was introduced
![2](https://user-images.githubusercontent.com/45123194/144712099-f2ef5722-2d83-4fc3-b4b4-ebb25ec97696.png)
![1](https://user-images.githubusercontent.com/45123194/144712100-4e0730b9-ee51-402a-9e2c-7ce84da7747f.png)
.

BenNottelling commented on src/bb-themes/admin_default/html/layout_default.phtml:
```diff
@@ -24,95 +24,156 @@
+<script>
+function openNav() {
+  document.getElementById("mySidebar").style.width = "220px";
+  document.getElementById("main").style.marginLeft = "220px";
```
There is no element with the ID of "main". Please delete this line

BenNottelling commented on src/bb-themes/admin_default/html/layout_default.phtml:
```diff
@@ -24,95 +24,156 @@
+
+function closeNav() {
+  document.getElementById("mySidebar").style.width = "0";
+  document.getElementById("main").style.marginLeft= "0";
```
Ditto

BenNottelling commented on src/bb-themes/admin_default/html/layout_default.phtml:
```diff
@@ -24,95 +24,156 @@
+<script>
+function openNav2() {
+  document.getElementById("mySidebar2").style.width = "200px";
+  document.getElementById("main2").style.marginRight = "200px";
```
Ditto

BenNottelling commented on src/bb-themes/admin_default/html/layout_default.phtml:
```diff
@@ -24,95 +24,156 @@
+
+function closeNav2() {
+  document.getElementById("mySidebar2").style.width = "0";
+  document.getElementById("main2").style.marginRight= "0";
```
Ditto

BenNottelling reviewed (changes requested):
Please address the couple of comments and resolve the issue with the sidebar scrolling with the page, it happens on both mobile and desktop, but it is worse on mobile. 
Thank you :) 


![Animation](https://user-images.githubusercontent.com/17304943/144720122-58c28cee-0963-4827-98f3-4bd259c56f18.gif)

BenNottelling reviewed (approved):
Per conversation with xSido, the sidebar needs to be scrollable (either separately or with the entire page) so that all items can be accessed if there are many of them.
I am OK with the current state of it.

@evrifaessa I know you mentioned not liking the sidebar, what are you looking to have improved?

evrifaessa commented:
- Why do we use the same menu icon for two different sidebars?
- Based on screenshots, I actually like the old sidebar better. The new one just looks like a list of links with no styling.
- CodeFactor found 400+ issues. Haven't checked them individually yet though.

xSido commented:
> * Why do we use the same menu icon for two different sidebars?
Because this is the first step in redesigning the admin panel. I doubt there will be anyone doing the entire work and then publish it in 1 commit. Especially when it need to be approved and it might not pass.

> * CodeFactor found 400+ issues. Haven't checked them individually yet though.
yes, there are some mistakes introduced by me.
6x  Unexpected unit (length-zero-no-unit)  while it worked the way i made, will correct this little problem.
3x Unexpected empty block - because i wanted to implement in future something there, but will remove them for now then.

but rest of them are from existing code.
most of them duplicated selector, maybe for cleaner look.

timothygwebb reviewed (changes requested):
A lot of the code factor issues are due to the min.css file. This has been discussed prior and their is an auto fix available for it. Please apply the auto fix.  I would also like to discuss in more detail the request to create this to be mobile compliant. I would like to be involved in this process and I understand responsive quite well and the requirements for device identification. I am looking forward to this redesign it does seem to be more user friendly. Outside of that if any one has any further questions or comments relating to merging this request please keep me in the loop. If you need assistance with the minimization of the min.css file please let me know as well.

AndPavlenko commented:
Seems like stoped work on this PR. I'd like to help finish redesign admin theme with migrate to SCSS, remove browser prefixes prefer Autoprefixer and build frontend with Npm. All like most modern projects. ;)
What do you think about it?

evrifaessa commented:
> Seems like stoped work on this PR. I'd like to help finish redesign admin theme with migrate to SCSS, remove browser prefixes prefer Autoprefixer and build frontend with Npm. All like most modern projects. ;) What do you think about it?

Contributor activity is pretty much dead at this point but sure, why not? I think a complete frontend rewrite might boost the interest in the project and give it a completely new look. I'm looking forward to it.

NiemandNatural commented:
Just give me a few PHP (or HTML) and (S)CSS files so I can change theming as I require it. I might throw out a few simple themes. For now, I have not yet read through the theming part of the docs - but judging the files by a first glance, it doesn't look as straight forward as I'd like it to be.

Also: Admin BackEnd needs a DarkMode. Really important! (I am using DarkReader but that's just looking urghs.)

It would also really help if not everything was on github - as, for example, the linked extensions.boxbilling.org seems down. Oh and if there was a few better explanations/previews (for mods and stuff). ;)

evrifaessa commented:
I know the backend sucks, and I don't think we can do anything with it in the short term. Everything is under `bb-themes`.

BenNottelling commented:
@xSido do you have any intentions of continuing work on this PR?

timothygwebb commented:
@andpavlenko @xSido This branch has conflicts that must be resolved
Use the [web editor](https://github.com/boxbilling/boxbilling/pull/1142/conflicts) or the  command line  to resolve conflicts.
Conflicting files
src/bb-themes/admin_default/html/layout_default.phtml
src/bb-themes/admin_default/html/layout_login.phtml
src/bb-themes/admin_default/html/mod_client_index.phtml
src/bb-themes/admin_default/html/mod_invoice_index.phtml
src/bb-themes/admin_default/html/mod_invoice_transactions.phtml
src/bb-themes/admin_default/html/mod_order_index.phtml
src/bb-themes/admin_default/html/mod_support_public_tickets.phtml
src/bb-themes/admin_default/html/mod_support_tickets.phtml
